layers/meta-balena-genericx86/recipes-support/hostapp-update-hooks/files/999-resin-boot-cleaner

```diff
@@ -28,15 +29,30 @@ if [ "$DURING_UPDATE" = "1" ]; then
 
 	# make sure the bootstrap code (boot.img) is removed in case we are using EFI boot
 	if [ -d /sys/firmware/efi ] ; then
-		device="/dev/"$(findmnt --noheadings --canonicalize --output SOURCE "$BALENA_BOOT_MOUNTPOINT" | xargs lsblk -no pkname)
-		dd if=/dev/zero of=$device bs=446 count=1
-
-		# re-add the EFI entry for resinOS boot from internal media as some EFI firmwares are buggy and won't detect the old entry anymore
-		# first remove existing resinOS entries so we won't have duplicates
-		printf "[INFO] Re-add EFI boot entry for starting resinOS from internal media.\n"
-		mount -t efivarfs efivarfs /sys/firmware/efi/efivars
-		duplicates=`efibootmgr | grep resinOS |sed 's/Boot*//g' | sed 's/* resinOS//g'`
-		for i in $duplicates; do efibootmgr -B -b $i; done
-		efibootmgr -c -d $device -p 1 -L "resinOS" -l "\EFI\BOOT\bootx64.efi"
+		boot_dev=$(findmnt --noheadings --canonicalize --output SOURCE "$BALENA_BOOT_MOUNTPOINT")
+		type=$(lsblk -no TYPE "${boot_dev}")
+		if [ "${type}" = "crypt" ]; then
+			boot_part="$(lsblk -sJ "${boot_dev}" | jq -r '.blockdevices[].children[].name')"
+		else
+			boot_part="$(lsblk -sJ "${boot_dev}" | jq -r '.blockdevices[].name')"
+		fi
+		device=$(lsblk "/dev/$boot_part" -dnlo PKNAME)
+		pttype=$(get_part_table_type "${device}")
+
+		if [ "${pttype}" = "mbr" ]; then
```

Comment:
Why is this done only for MBR partitions? Even if this device type does not use GPT partitions we should consider it. I am wondering whether this hook that is used in several device types should be moved to meta-balena. What do you think?

Reply:
@mtoman as discussed, let's do `generic-amd64` as it's a better fit. I don't specially like having two different partition tables used in the same DT.